src/physics/large_eddy_simulation.h

```diff
@@ -0,0 +1,285 @@
+        const dealii::types::global_dof_index cell_index) const = 0;
+
+protected:
+    /// Returns the square of the magnitude of the tensor (i.e. the double dot product of a tensor with itself)
+    template<typename real2> 
+    real2 get_tensor_magnitude_sqr (const dealii::Tensor<2,dim,real2> &tensor) const;
+
+    /// Templated dissipative (i.e. viscous) flux: \f$ \mathbf{F}_{diss} \f$ 
+    template<typename real2>
+    std::array<dealii::Tensor<1,dim,real2>,nstate> dissipative_flux_templated (
+        const std::array<real2,nstate> &conservative_soln,
+        const std::array<dealii::Tensor<1,dim,real2>,nstate> &solution_gradient,
+        const dealii::types::global_dof_index cell_index) const;
+
+    /** Dissipative flux Jacobian (repeated from NavierStokes)
+     *  Note: Only used for computing the manufactured solution source term;
+     *        computed using automatic differentiation
+     */
+    dealii::Tensor<2,nstate,real> dissipative_flux_directional_jacobian (
+        const std::array<real,nstate> &conservative_soln,
+        const std::array<dealii::Tensor<1,dim,real>,nstate> &solution_gradient,
+        const dealii::Tensor<1,dim,real> &normal,
+        const dealii::types::global_dof_index cell_index) const;
+
+    /** Dissipative flux Jacobian wrt gradient component (repeated from NavierStokes)
+     *  Note: Only used for computing the manufactured solution source term;
+     *        computed using automatic differentiation
+     */
+    dealii::Tensor<2,nstate,real> dissipative_flux_directional_jacobian_wrt_gradient_component (
+        const std::array<real,nstate> &conservative_soln,
+        const std::array<dealii::Tensor<1,dim,real>,nstate> &solution_gradient,
+        const dealii::Tensor<1,dim,real> &normal,
+        const int d_gradient,
+        const dealii::types::global_dof_index cell_index) const;
+
+    /// Get manufactured solution value (repeated from Euler)
+    std::array<real,nstate> get_manufactured_solution_value (
+        const dealii::Point<dim,real> &pos) const;
+
+    /// Get manufactured solution value (repeated from Euler)
+    std::array<dealii::Tensor<1,dim,real>,nstate> get_manufactured_solution_gradient (
+        const dealii::Point<dim,real> &pos) const;
+
+    /// Dissipative flux contribution to the source term (repeated from NavierStokes)
+    std::array<real,nstate> dissipative_source_term (
+        const dealii::Point<dim,real> &pos,
+        const dealii::types::global_dof_index cell_index) const;
+};
+
+/// Smagorinsky eddy viscosity model. Derived from Large Eddy Simulation.
+template <int dim, int nstate, typename real>
+class LargeEddySimulation_Smagorinsky : public LargeEddySimulationBase <dim, nstate, real>
+{
+public:
+    using thermal_boundary_condition_enum = Parameters::NavierStokesParam::ThermalBoundaryCondition;
+    /** Constructor for the sub-grid scale (SGS) model: Smagorinsky
+     *  Reference: plata2019performance
```

Comment:
Googling this reference doesn't lead to the model's paper. Maybe include a link to a book/paper used for the equations used.

Reply:
fixed